base/collection_xattr_common.go

```diff
@@ -192,19 +198,23 @@ func WriteUpdateWithXattr(store SubdocXattrStore, k string, xattrKey string, use
 	var cas uint64
 	emptyCas := uint64(0)
 
-	// If an existing value has been provided, use that as the initial value
-	if previous != nil && previous.Cas > 0 {
-		value = previous.Body
-		xattrValue = previous.Xattr
-		cas = previous.Cas
-		userXattrValue = previous.UserXattr
-	}
-
 	for {
 		var err error
-		// If no existing value has been provided, retrieve the current value from the bucket
-		if cas == 0 {
-			// Load the existing value.
+		if previous != nil {
+			// If an existing value has been provided, use that as the initial value.
+			// A zero CAS is interpreted as no document existing.
+			if previous.Cas != 0 {
+				value = previous.Body
+				xattrValue = previous.Xattr
+				cas = previous.Cas
+				userXattrValue = previous.UserXattr
+			} else if previous.Body != nil || previous.Xattr != nil {
+				panic("WriteUpdateWithXattr: previous doc has zero Cas but is not empty")
```

Comment:
I'll take this out -- IIRC this was there because I wasn't sure if something would legitimately pass a `previous` with a zero CAS, and if so this would let me find out when running tests.